src/NHibernate/Mapping/ByCode/Impl/CustomizersImpl/PropertyContainerCustomizer.cs

```diff
@@ -118,6 +149,34 @@ protected virtual void RegisterDynamicComponentMapping<TComponent>(Expression<Fu
 			RegisterDynamicComponentMapping(mapping, member, memberOf);
 		}
 
+		protected virtual void RegisterDynamicComponentMapping<TComponent>(
+			Expression<Func<TEntity, IDictionary>> property,
+			IDictionary<string, System.Type> template,
+			Action<IDynamicComponentMapper<TComponent>> mapping)
+		{
+			MemberInfo member = TypeExtensions.DecodeMemberAccessExpression(property);
+			MemberInfo memberOf = TypeExtensions.DecodeMemberAccessExpressionOf(property);
+			RegisterDynamicComponentMapping(template, mapping, member, memberOf);
+		}
+
+		protected virtual void RegisterDynamicComponentMapping<TComponent>(
+			Expression<Func<TEntity, IDictionary<string, object>>> property,
+			IDictionary<string, System.Type> template,
+			Action<IDynamicComponentMapper<TComponent>> mapping)
+		{
+			MemberInfo member = TypeExtensions.DecodeMemberAccessExpression(property);
+			MemberInfo memberOf = TypeExtensions.DecodeMemberAccessExpressionOf(property);
+			RegisterDynamicComponentMapping(template, mapping, member, memberOf);
+		}
+
+		protected void RegisterDynamicComponentMapping<TComponent>(IDictionary<string, System.Type> template, Action<IDynamicComponentMapper<TComponent>> mapping, params MemberInfo[] members)
+		{
+			foreach (var member in members)
+			{
+				mapping(new DynamicComponentCustomizer<TComponent>(template, explicitDeclarationsHolder, CustomizersHolder, new PropertyPath(PropertyPath, member)));
```

Comment:
Here, inside `new DynamicComponentCustomizer<TComponent>(...)` for every member we call `CreateType()`